log/log.go

```diff
@@ -79,11 +79,11 @@ func Warn(ctx context.Context, event string, opts ...option) {
 	eventFuncInst.f(ctx, event, WARN, opts...)
 }
 
-func ErrorDetails(ctx context.Context, event string, opts ...option) {
+func ErrorDetails(ctx context.Context, event string, err error, opts ...option) {
 	eventFuncInst.f(ctx, event, ERROR, opts...)
 }
 
-func Fatal(ctx context.Context, event string, opts ...option) {
+func Fatal(ctx context.Context, event string, err error, opts ...option) {
 	eventFuncInst.f(ctx, event, FATAL, opts...)
```

Comment:
Atm the `err` isn't being used by the functions and it wouldn't appear in the logs. We would want to add it to the `eventFuncInst.f()` call. Perhaps we could add some intermediary logic before the call to `f()` to append the `err` to `opts` so that the error can then be displayed in the logs?